pkg/vpcmodel/explainabilityInput.go

```diff
@@ -0,0 +1,226 @@
+	if err1 != nil {
+		return nil, false, err1
+	}
+	if vsi != nil {
+		return vsi, false, nil
+	}
+	// cidrOrName, if legal, references an address.
+	// 2. cidrOrName references an ip address
+	return c.getNodesFromAddress(cidrOrName)
+}
+
+// getNodesOfVsi gets a string name or UID of VSI, and
+// returns the list of all nodes within this vsi
+func (c *VPCConfig) getNodesOfVsi(vsi string) ([]Node, error) {
+	var nodeSetWithVsi NodeSet
+	for _, nodeSet := range c.NodeSets {
+		// todo: at the moment we consider here all NodeSets and not just vsis (e.g. also subnets)
+		//       fix once we have abstract vpc and subnets (#380)
+		if nodeSet.Name() == vsi || nodeSet.UID() == vsi {
+			if nodeSetWithVsi != nil {
+				return nil, fmt.Errorf("there is more than one resource (%s, %s) with the given input string %s representing its name. "+
+					"can not determine which resource to analyze. consider using unique names or use input UID instead",
+					nodeSetWithVsi.UID(), nodeSet.UID(), vsi)
+			}
+			nodeSetWithVsi = nodeSet
+		}
+	}
+	if nodeSetWithVsi == nil {
+		return nil, nil
+	}
+	return nodeSetWithVsi.Nodes(), nil
+}
+
+// getNodesFromAddress gets a string that should present a cidr or IP
+// and returns the corresponding node(s) and a bool which is true iff cidrOrName is an internal address
+// (and the nodes are its network interfaces). Specifically:
+//  1. If it does not present a cidr or IP, returns nil and false
+//  2. If it represents a cidr which is both internal and external, returns an error
+//  3. If it presents an external address, returns external addresses nodes and false
+//  4. If it contains internal address not within the address prefix of this vpc, returns an error
+//  5. If it presents an internal address, return connected network interfaces if any and true,
+//     error otherwise
+func (c *VPCConfig) getNodesFromAddress(ipOrCidr string) (nodes []Node, internalIP bool, err error) {
+	inputIPBlock := common.NewIPBlockFromCidrOrAddress(ipOrCidr)
+	// 1.
+	if inputIPBlock == nil { // 1. string cidr does not represent a legal cidr
+		return nil, false, nil
+	}
+	// 2.
+	_, publicInternet, err1 := getPublicInternetIPblocksList()
+	if err1 != nil {
+		return nil, false, err1
+	}
+	// 2.
+	isExternal := !inputIPBlock.Intersection(publicInternet).Empty()
+	isInternal := !inputIPBlock.ContainedIn(publicInternet)
+	if isInternal && isExternal {
+		return nil, false, fmt.Errorf("%s contains external and internal addresses which is not supported. "+
+			"src, dst should be external *or* internal address", ipOrCidr)
+	}
+	// 3.
+	if isExternal { // 3.
+		nodes, err = c.getCidrExternalNodes(inputIPBlock)
+		return nodes, false, err
+		// internal address
+	} else if isInternal {
+		// 4.
+		vpcAP := c.VPC.AddressRange()
+		if !inputIPBlock.ContainedIn(vpcAP) {
+			return nil, false, fmt.Errorf("internal address %s not within vpc's address prefix %s",
+				inputIPBlock.ToIPRanges(), vpcAP.ToIPRanges())
+		}
+		// 5.
```

Comment:
`VPC.AddressRange()` returns the ranges for VPC's subnets and not the address prefix.

Reply:
as agreed, we'll change the relevant error msg add a todo (better have address prefix, but at the moment not worth the effort)